fileManagment/ViewController.swift

```diff
@@ -0,0 +1,184 @@
+//
+//  ViewController.swift
+//  fileManagment
+//
+//  Created by Rayan Taj on 23/11/2021.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    // UI elements
+    @IBOutlet weak var filesTextField: UITextField!
+    @IBOutlet weak var typeFileSwitch: UISwitch!
+    @IBOutlet weak var fileTableView: UITableView!
+    @IBOutlet weak var contentTextField: UITextView!
+    @IBOutlet weak var actionTypeSegment: UISegmentedControl!
+    
+    // The file manager object is resposible for communicating with files in the system
+    let fileManager = FileManager.default
+    
+    var  appFilesDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first
+    
+    // This array will contain all existing folders
+    var foldersArray :[String] = []
+    var  selectedFolderName : String? =  nil
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // This function will assign all existing folders to the array
+        relodFiles()
+ 
+    }
+    
+    
+    func relodFiles()  {
+                
+        do {
+            
+            let fileURLs = try fileManager.contentsOfDirectory(atPath: appFilesDirectory!.path)
+            
+            foldersArray.removeAll()
+            
+            for folder in fileURLs {
+                let test = appFilesDirectory!.appendingPathComponent(folder)
+                if test.hasDirectoryPath {
+                    foldersArray.append(folder)
+                }
+                 
+            }
+            
+            
+        } catch {
+            print("Error while enumerating files \(appFilesDirectory!.path): \(error.localizedDescription)")
+        }
+    }
+    
+    
+    @IBAction func createFileAction(_ sender: Any) {
+        
+let typeIndex = actionTypeSegment.selectedSegmentIndex
+   
+        
+                if let name = filesTextField.text {
+            do {
+                
+                switch typeIndex {
+                    
+                case 0 :
```

Comment:
When reading your code, how do you know 0 is meant for file creation without reading the rest of the code?
an idea to improve your code you might use [enumerations](https://docs.swift.org/swift-book/LanguageGuide/Enumerations.html) raw value! 
How? 
This is just an example, you may write:
```swift
enum Create: Int {
    case file = 0, folder = 1
}
```
and then call it in your switch, for example you'll write `case Create.file.rawValue : `  instead of `case 0 :` 
which is more readable.

Reply:
that's true, am gonna implement it